src/runtime/virtcontainers/qemu_arch_base.go

```diff
@@ -318,24 +319,50 @@ func (q *qemuArchBase) memoryTopology(memoryMb, hostMemoryMb uint64, slots uint8
 }
 
 func (q *qemuArchBase) appendConsole(_ context.Context, devices []govmmQemu.Device, path string) ([]govmmQemu.Device, error) {
-	serial := govmmQemu.SerialDevice{
-		Driver:        govmmQemu.VirtioSerial,
-		ID:            "serial0",
-		DisableModern: q.nestedRun,
-		MaxPorts:      uint(2),
-	}
+	var serial, console govmmQemu.Device
+	var consoleKernelParams []Param
 
-	devices = append(devices, serial)
+	if q.legacySerial {
+		serial = govmmQemu.LegacySerialDevice{
+			Chardev: "charconsole0",
+		}
+
+		console = govmmQemu.CharDevice{
+			Driver:   govmmQemu.LegacySerial,
+			Backend:  govmmQemu.Socket,
+			DeviceID: "console0",
+			ID:       "charconsole0",
+			Path:     path,
```

Comment:
This path is implicitly defined right? Maybe in the future we can treat this as a regular caller added device where the caller can chose both the Backend type as well as the path. That would make consuming this in other projects explicit. But that is for a later day.

Reply:
Indeed, I don'r recall that it is configurable ATM